metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -168,16 +161,18 @@ class MetalsLanguageServer(
       initializeParams = Option(params)
       updateWorkspaceDirectory(params)
       val capabilities = new ServerCapabilities()
-      capabilities.setCompletionProvider(
-        new CompletionOptions(false, Collections.emptyList())
-      )
-      capabilities.setCompletionProvider(
-        new CompletionOptions(false, Collections.emptyList())
-      )
       capabilities.setDefinitionProvider(true)
       capabilities.setTextDocumentSync(TextDocumentSyncKind.Full)
       if (config.isNoInitialized) {
-        initialized(new InitializedParams).asScala
+        sh.schedule(
```

Comment:
Updated. SAM sure is prettier, and for dealing with Java collections they're almost a necessity (I can never get the covariant/contravarian bounds right for `java.util.function.Function`).

Reply:
Sure would be nice to have  a Scalafix rewrite to fix these automatically.

As for 2.11 support, I would rather explore classloading 2.11 presentation compiler from a 2.12 server instead of compiling the server to 2.11.